controllers/machineset_delete_policy.go

```diff
@@ -53,6 +55,10 @@ func oldestDeletePriority(machine *clusterv1.Machine) deletePriority {
 	if machine.Status.FailureReason != nil || machine.Status.FailureMessage != nil {
 		return mustDelete
 	}
+	nodeHealthyCondition := conditions.Get(machine, clusterv1.MachineNodeHealthyCondition)
```

Comment:
```
if machine.Status.NodeRef == nil {
if machine.Status.FailureReason != nil
if machine.Status.FailureMessage != nil {
if nodeHealthyCondition != nil && nodeHealthyCondition.Status != corev1.ConditionTrue {
```
Is now duplicated across all policies and have a common semantic with a unique deletePriority, can we please create a unit tested function e.g `isMachineHealthy` and reuse it across policies?

Reply:
@moonek does this make sense to you ^^?

Reply:
@enxebre It doesn't seem difficult to create a common function in controller source, but I'm not familiar with writing unit tests. Do you want to be reflected in this PR?

Reply:
hey @moonek yes please let's address it in this PR.
For the unit tests in addition to the changes you already include in this PR we'll just need to include a new func e.g.

```
func TestIsMachineHealthy(t *testing.T) {
	testCases := []struct {
		name     string
		machine  clusterv1.Machine
		expected bool
	}{
		{
			name:     "when it has no NodeRef",
			machine:  clusterv1.Machine{},
			expected: false,
		},
		{
			name:     "when it has a FailureReason",
			machine:  clusterv1.Machine{},
			expected: false,
		},
		{
			name:     "when it has a FailureMessage",
			machine:  clusterv1.Machine{},
			expected: false,
		},
		{
			name:     "when nodeHealthyCondition is not false",
			machine:  clusterv1.Machine{},
			expected: false,
		},
		{
			name:     "when nodeHealthyCondition is not unknown",
			machine:  clusterv1.Machine{},
			expected: false,
		},
		{
			name:     "when al requirements are met for node to be healthy",
			machine:  clusterv1.Machine{},
			expected: true,
		},
	}

	for _, test := range testCases {
		t.Run(test.name, func(t *testing.T) {
			g := NewWithT(t)
			g.Expect(isMachineHealthy(test.machine)).To(Equal(test.expected))
		})
	}
}
```
making sure each machine matches the test case.

Reply:
@enxebre It helped a lot. Please review the changed commit.